lib/models/watcher.js

```diff
@@ -16,17 +18,65 @@ class Watcher extends CoreObject {
     logger.info('initialize %o', options);
 
     this.serving = _options.serving;
-    this.watcher = this.watcher || this.constructWatcher(options);
+
+    if (isExperimentEnabled('BROCCOLI_WATCHER')) {
+      this.watcher = this.watcher || this.constructBroccoliWatcher(options);
+    } else {
+      // This branch can be removed once BROCCOLI_WATCHER is enabled by default
+      this.watcher = this.watcher || this.constructWatcher(options);
+      this.watcher.on('change', this.didChange.bind(this));
+    }
 
     this.watcher.on('error', this.didError.bind(this));
-    this.watcher.on('change', this.didChange.bind(this));
     this.serveURL = serveURL;
   }
 
   constructWatcher(options) {
     return new (require('ember-cli-broccoli-sane-watcher'))(this.builder, options);
   }
 
+  constructBroccoliWatcher(options) {
+    let watcher = new (require('broccoli')).Watcher(this.builder, this.builder.builder.watchedSourceNodeWrappers, {
```

Comment:
this would be much easier to read if we separate the require from the instantiation

```suggestion
const { Watcher } = require('broccoli');
let watchedNodes = this.builder.builder.watchedSourceNodeWrappers
let watcher = new Watcher(this.builder, watchedNodes, {
```

Reply:
I agree and updated